source/server/feature_toggles.cpp

```diff
@@ -0,0 +1,12 @@
+#include "feature_toggles.h"
+
+namespace nidevice_grpc {
+bool FeatureToggles::is_enabled(const std::string& feature_name) const
+{
+  auto found = map_.find(feature_name);
+  if (found != map_.end()) {
+    return found->second;
+  }
+  return false;
```

Comment:
Returning a simple bool makes it difficult to distinguish from a feature that is explicitly toggled to false and one that isn't  specified. In the latter case, you'd want in-progress features to default to false and released features to true.

Reply:
I was thinking that adding the concept of declaring features and readiness would be a future enhancement. And then this would still boil down to a boolean that takes all that into account. But, yeah, in the current setup a tri-state is better. So I'll go with that. I hope that the readiness thing goes in, but it's not a current priority. So better be ready for it to be stuck as is.